src/plugins/orangehrmBuzzPlugin/Dao/BuzzDao.php

```diff
@@ -414,4 +414,22 @@ public function getBuzzShareByPostId(int $postId): ?BuzzShare
 
         return $q->getQuery()->getOneOrNullResult();
     }
+
+    /**
+     * @param int $shareId
+     * @return BuzzShare[]
+     */
+    public function getBuzzPostSharesById(int $shareId): array
+    {
+        /** @var BuzzShare $share */
+        $share = $this->getRepository(BuzzShare::class)->find($shareId);
+
+        if ($share->getType() === BuzzShare::TYPE_SHARE) {
```

Comment:
no need to handle it here. if provided id is a share, then throw bad request from the API

Reply:
since FE will send post id removed checking for type share